rectools/models/nn/transformers/base.py

```diff
@@ -427,6 +427,31 @@ def _custom_transform_dataset_i2i(
     ) -> Dataset:
         return self.data_preparator.transform_dataset_i2i(dataset)
 
+    def _fit_partial(self, dataset: Dataset, epochs: int) -> None:
+        if not self.is_fitted:
+            self.data_preparator.process_dataset_train(dataset)
+
+            item_model = self._construct_item_net(self.data_preparator.train_dataset)
+            torch_model = self._init_torch_model(item_model)
+
+            dataset_schema = self.data_preparator.train_dataset.get_schema()
+            item_external_ids = self.data_preparator.train_dataset.item_id_map.external_ids
+            model_config = self.get_config(simple_types=True)
+            self._init_lightning_model(
+                torch_model=torch_model,
+                dataset_schema=dataset_schema,
+                item_external_ids=item_external_ids,
+                model_config=model_config,
```

Comment:
Let's create `_build_model_from_dataset` method to handle this logic both in _fit and _fit_partial to stop duplicating code.